src/hera/workflow.py

```diff
@@ -56,14 +59,17 @@ def __init__(
         labels: Optional[Dict[str, str]] = None,
         namespace: Optional[str] = None,
         security_context: Optional[WorkflowSecurityContext] = None,
+        workflow_template_ref: Optional[str] = None,
     ):
         self.name = f'{name.replace("_", "-")}-{str(uuid4()).split("-")[0]}'  # RFC1123
+        self.name_original = name
```

Comment:
I am happy this was contributed in this PR as well, because it indicates there's a desire for user submitted names to be used as they are specified, rather than Hera modify the name to avoid collisions on behalf of the user and then submit the workflow. I suggested we introduce this change in 3.0.0, which I can release soon. Any chance this introduction can be removed from this PR for now, until it's released in 3.0.0? 🙂

Reply:
Done. Removed with the latest commit. I hope now everything is looking good.